src/providers/BackgroundBlurProvider/index.tsx

```diff
@@ -0,0 +1,147 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+  BackgroundBlurOptions,
+  BackgroundBlurVideoFrameProcessor,
+  BackgroundFilterSpec,
+  ConsoleLogger,
+  DefaultVideoTransformDevice,
+  Device,
+  LogLevel,
+  NoOpVideoFrameProcessor,
+  VideoFrameProcessor,
+} from 'amazon-chime-sdk-js';
+import React, {
+  createContext,
+  FC,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+
+interface Props {
+  /** The spec defines the assets that will be used for adding background blur to a frame */
+  spec?: BackgroundFilterSpec;
+  /** A set of options that can be supplied when creating a background blur video frame processor. */
+  options?: BackgroundBlurOptions;
+}
+
+interface BackgroundBlurProviderState {
+  createBackgroundBlurDevice: (
+    device: Device
+  ) => Promise<DefaultVideoTransformDevice>;
+  isBackgroundBlurSupported: boolean | undefined;
+}
+
+const BackgroundBlurProviderContext =
+  createContext<BackgroundBlurProviderState | null>(null);
+
+const BackgroundBlurProvider: FC<Props> = ({ spec, options, children }) => {
+  const [isBackgroundBlurSupported, setIsBackgroundBlurSupported] = useState<
+    boolean | undefined
+  >(undefined);
+  const [processor, setProcessor] = useState<VideoFrameProcessor | undefined>();
+
+  useEffect(() => {
+    async function initializeBackgroundBlur() {
+      try {
+        console.log('Initializing background blur processor.');
+        const createdProcessor = await BackgroundBlurVideoFrameProcessor.create(
+          spec,
+          options
+        );
+
+        if (createdProcessor instanceof NoOpVideoFrameProcessor) {
+          console.log(`Initialized NoOpVideoFrameProcessor.`);
+          setProcessor(undefined);
+          setIsBackgroundBlurSupported(false);
+        } else {
+          console.log(
+            `Initialized background blur processor: ${JSON.stringify(
+              processor
+            )}`
+          );
+          setProcessor(createdProcessor);
+          setIsBackgroundBlurSupported(true);
```

Comment:
[For learning]: The background blur processor has the isSupported method, so should not the support be based on it and not only the successful creation, also should the creation be based on the same isSupported API?